packages/apputils-extension/src/index.ts

```diff
@@ -549,18 +669,20 @@ namespace Private {
    *
    * @param ready - A promise that must be resolved before splash disappears.
    *
-   * @param recovery - A function that recovers from a hanging splash.
+   * @param recovery - A command that recovers from a hanging splash.
    */
   export
-  function showSplash(ready: Promise<any>, recovery: () => void): IDisposable {
+  function showSplash(ready: Promise<any>, commands: CommandRegistry, recovery: string): IDisposable {
     splash.classList.remove('splash-fade');
     splashCount++;
 
     if (debouncer) {
       window.clearTimeout(debouncer);
     }
     debouncer = window.setTimeout(() => {
-      recover(recovery);
+      if (commands.hasCommand(recovery)) {
```

Comment:
Is the reason for this check that the command may not have been added yet?

Reply:
If you generate the pre-startup "pick a new workspace" error state, then the splash screen may be up for a long time and it will trigger the splash screen recovery function. But because the page has not finished loading, the recovery command will not be available. This `hasCommand()` check basically turns the `recover()` function into a no-op in that case.